src/gausskernel/cbb/instruments/statement/instr_statement.cpp

```diff
@@ -1418,6 +1420,10 @@ void instr_stmt_report_basic_info()
         ResourceOwner tmpOwner = t_thrd.utils_cxt.CurrentResourceOwner;
         t_thrd.utils_cxt.CurrentResourceOwner = old_cur_owner;
         ResourceOwnerDelete(tmpOwner);
+        if (to_update_db_name || to_update_user_name || to_update_client_addr) {
+            // start a new PG Gate session if any changes in db, user name, or client address
+            PgGate_InitSession(NULL, u_sess->statement_cxt.db_name);
```

Comment:
Do you expect this call to reinitialize a session? I was planning on making InitSession sticky - once initialized, all subsequent calls would be noop

Reply:
When any one of database name, user name, or client address changes, it would be a different database session even the connection could be landed on the same thread in the thread pool. We should reinitialize the session in this case, right?

Reply:
Discussed offline. The implication to SKV is that we have to abort any ongoing txn in the thread which calls PGInit.

Reply:
after discussion, we only need to remember the database name in the session object in pggate layer. K2 connection does not need to be changed. We might need to abort the on-going transaction if this happens.